src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -179,20 +183,31 @@ class JesBackend extends Backend with LazyLogging {
       taskOutput.name -> rawValue
     }.toMap
 
-    status match {
-      case Run.Success(created, started, finished) =>
-        val taskOutputEvaluationFailures = outputMappings.filter {_._2.isFailure}
-        if (taskOutputEvaluationFailures.isEmpty) {
-          val unwrappedMap = outputMappings.collect { case (name, Success(wdlValue) ) => name -> wdlValue }
-          Success(unwrappedMap)
-        } else {
-          val message = taskOutputEvaluationFailures.collect { case (name, Failure(e))  => s"$name: $e" }.mkString("\n")
-          Failure(new Throwable(s"Workflow ${workflowDescriptor.id}: $message"))
-        }
-      case Run.Failed(created, started, finished, errorCode, errorMessage) =>
-        Failure(new Throwable(s"Workflow ${workflowDescriptor.id}: errorCode $errorCode for command: $commandLine. Message: $errorMessage"))
+    val warnAboutStderrLength: BigInteger = if(!call.failOnStderr) {
+      BigInteger.valueOf(0)
+    } else {
+      JesConnection.storage.objectSize(GoogleCloudStoragePath(stdErrJesOutput(callGcsPath).gcs))
     }
 
+    if(warnAboutStderrLength.intValue > 0) {
+      Failure(new Throwable(s"Workflow ${workflowDescriptor.id}: stderr has length $warnAboutStderrLength for command: $instantiatedCommandLine"))
+    } else {
+      status match {
+        case Run.Success(created, started, finished) =>
+          val taskOutputEvaluationFailures = outputMappings.filter {
```

Comment:
IMO these little short higher order functions read better when a single line. 

Picture the ideal situation as being a chain of operations that you can read left to right almost like english. Obviously there are cases where that goes on for too long or it is too complicated, and that's when you don't do things that way.